server/src/main/webapp/WEB-INF/rails/webpack/views/pages/pipeline_activity/build_cause_widget.tsx

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2019 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {bind} from "classnames/bind";
+import {MithrilViewComponent} from "jsx/mithril-component";
+import m from "mithril";
+import Stream = require("mithril/stream");
+import {MaterialRevision, Modification, PipelineRunInfo} from "models/pipeline_activity/pipeline_activity";
+import {Dropdown, DropdownAttrs} from "views/components/buttons";
+import * as Icons from "views/components/icons";
+import styles from "./index.scss";
+
+const classnames = bind(styles);
+
+interface Attrs {
+  pipelineRunInfo: PipelineRunInfo;
+  showBuildCaseFor: Stream<string>;
+}
+
+export class BuildCauseWidget extends Dropdown<Attrs> {
+  toggleDropdown(vnode: m.Vnode<DropdownAttrs & Attrs>, e: MouseEvent) {
+    super.toggleDropdown(vnode, e);
+    if (vnode.attrs.show()) {
+      vnode.attrs.showBuildCaseFor(vnode.attrs.pipelineRunInfo.counterOrLabel());
+    } else {
+      vnode.attrs.showBuildCaseFor("");
+    }
+  }
+
+  protected doRenderButton(vnode: m.Vnode<DropdownAttrs & Attrs>): m.Children {
+    if (!vnode.attrs.pipelineRunInfo.revision()) {
+      return;
+    }
+
+    return <a class={styles.buildCauseButton}
+              data-test-id="trigger-with-changes-button"
```

Comment:
It is not actually a button rather a link. So can we use `trigger-with-changes-link` as a data-test-id?